docs: Update "deployments using Vercel (formerly ZEIT Now)" (#1195) (#1199)
sveltejs/sapper: site/content/docs/11-deploying.md

Sapper apps run anywhere that supports Node 8 or higher.

### Deploying to Vercel ([formerly ZEIT Now](https://vercel.com/blog/zeit-is-now-vercel))

> We can use an unoffical third-party libary like [vercel-sapper](https://www.npmjs.com/package/vercel-sapper) builder to deploy our projects to [Vercel] and even use our Sapper runtime environment there.

Reviewer: typo
```suggestion
We can use an unoffical third-party library like [vercel-sapper](https://www.npmjs.com/package/vercel-sapper) builder to deploy our projects to [Vercel] and even use our Sapper runtime environment there.
```

Sapper apps run anywhere that supports Node 8 or higher.

### Deploying to Vercel ([formerly ZEIT Now](https://vercel.com/blog/zeit-is-now-vercel))

> 

Reviewer 2: ```suggestion
```

> #### Using Vercel

Reviewer 2: ```suggestion
```

> We can use an unoffical third-party libary like [vercel-sapper](https://www.npmjs.com/package/vercel-sapper) builder to deploy our projects to [Vercel] and even use our sapper runtime environment there.

Reviewer 2: ```suggestion
We can use an unoffical third-party libary like [vercel-sapper](https://www.npmjs.com/package/vercel-sapper) builder to deploy our projects to [Vercel] and even use our Sapper runtime environment there.
```

Sapper apps run anywhere that supports Node 8 or higher.

### Deploying to Vercel ([formerly ZEIT Now](https://vercel.com/blog/zeit-is-now-vercel))

> #### Using Now v2

Reviewer 2: Shouldn't this say Vercel?

Reviewer 3: @benmccann product name is now, company name is Vercel :)

Reviewer 3: Okay, looks like I'm wrong. They seem to have renamed the product after the company. I'm so confused.

Author: I also am very confused about this.
The product is called "Vercel". [The CLI also uses this name after the update](https://vercel.com/docs/v2/build-step).
The [domains continue to use `.now.sh`](https://vercel.com/docs/v2/custom-domains) 

However I would say

> Link your GitHub account with your personal account or team on Vercel from your dashboard. [...] Then, click "Install Now for GitHub".

Makes `now` the winner for... now.
Opinions?

Author: Okay updated opinion:
It's likely that

> On the installation page for our command-line interface, the vercel package is now recommended instead of the legacy now package
[Source: Changelog May](https://vercel.com/blog/changelog-may-2020#dashboard)

has the answer.
I would say we should change the heading to `vercel` as the package for `now` seems to be legacy.

Then again, the third party library probably needs to be updated or forked to support vercel, but this is not a sapper task, so it should be seen asa different task unrelated to this change imo.

Reviewer 3: Yeah, I've just asked them on twitter.

I'm a maintainer of the third-party library so renaming that is no problem if @thgh agrees.

Reviewer 2: It looks like the Vercel documentation is outdated. I filed an issue with them: https://github.com/zeit/docs/issues/1842

> #### Using Now v2

Reviewer 2: I think we decided that this should say "Vercel" instead of "Now v2", right?

Reviewer 2: @diverent2 just checking in to see if you'd be able to update this line. That's the last comment I have on this PR. Thanks!

> #### Using Now v2

Reviewer 3: ```suggestion
#### Using Vercel
```

> We can use an unoffical third-party libary like [now-sapper loader](https://www.npmjs.com/package/now-sapper) to deploy our projects to [Vercel] and even use our sapper runtime environment there.

Reviewer 2: Why do we need to use an unofficial third-party library instead of doing `npm i -g vercel`?

Reviewer 4: Now (and also, I would assume, Vercel) does not support dynamic Sapper applications. They only natively support exported Sapper apps, the sort that you could host on any static file host. https://github.com/zeit/now-examples/issues/193#issuecomment-517802287 Thus the third-party library. I have not tried it, but my understanding is that it does not separate the different routes into their own files, which is something that a proper Now (Vercel?) builder ought to do.

Reviewer 3: @benmccann you still need the `vercel` cli regardless. `now-sapper` is a builder. As @Conduitry says `vercel/zeit/now/whatever it's called` have no explicit for Sapper apps, simply the static apps that Sapper can build.

`now-sapper` gives us that support, albeit with one (minor) tradeoff.

Reviewer 2: I don't quite understand why Vercel doesn't support Sapper by default. Is it because Sapper doesn't run on Lambda (https://github.com/sveltejs/sapper/issues/356)? If so, I sent a PR that I think would enable that: https://github.com/sveltejs/sapper/pull/1215.

Reviewer 2: > you still need the vercel cli regardless

We should document that. If I go to https://github.com/thgh/now-sapper it doesn't tell you how to deploy the project after it's built. It says how to configure the project, but not what commands to run to build or deploy

Anyway, I'm still not really understanding why a third-party tool is required. Vercel has a [Svelte sample](https://github.com/zeit/now/tree/master/examples/svelte) that includes server-side code. When I visit their deployed sample I can see that it does a `fetch` call in the Chrome debugger.

Maybe we should just link to [their instructions](https://vercel.com/guides/deploying-svelte-with-vercel) instead?

Reviewer 4: I hadn't seen that example before, but I do also notice that it is a _Svelte_ app, not a Sapper one. The server-side code is just individual files that CJS-export a `(req, res)` object to handle requests to that URL - it's not related at all to Svelte, which seems to be running entirely on the client side.

Doing a proper deployment to Now/Vercel of a dynamic SSR'd Sapper app would presumably involve generating a separate lambda for each page of the app, similar to the `api/date.js` in the Svelte example.

Reviewer 3: @benmccann their examples don't work for Sapper, that's why we built `now-sapper`. I don't know the technical details of it, but feel free to try to deploy a Sapper app without the builder if you understand how to diagnose the issues and document a simpler approach. You'll notice that `next` itself has a `now-next` builder, so writing a builder does seem to be the official approach.

With regards to deployment instructions, the view was that you'd find Now and then want a builder, so you'd know that the deployment for Vercel is as simple as typing three letters: `now`. Or I guess `vercel` today.

I agree, this is something we should document though.

Reviewer 2: Thanks for bearing with me on this and providing the additional info. This looks good to me. I sent a PR to update the README in the other project: https://github.com/thgh/now-sapper/pull/28

> We can use an unoffical third-party libary like [now-sapper loader](https://www.npmjs.com/package/now-sapper) to deploy our projects to [Vercel] and even use our sapper runtime environment there.

Reviewer 2: @diverent2 the package has been renamed to https://github.com/thgh/vercel-sapper. Mind updating it? (btw, I might remove the text "loader" from the link since it's not part of the repo name)

Visit the corresponding github repository of [vercel-sapper](https://github.com/thgh/vercel-sapper#readme) builder for implementation details and bug reports regarding [Vercel] deployments.

> > Note about Now v1

Reviewer 3: ```suggestion
```

> >

Reviewer 3: ```suggestion
```

> > Now v1 is deprecated and using it should be avoided, see [this comment](https://github.com/zeit/now/issues/1805#issuecomment-452470953).  

Reviewer 3: ```suggestion
```

> > If you are still running v1 consider upgrating to the newest version.

Reviewer 3: ```suggestion
```

const ASSETS = `cache${timestamp}`;

export default {
  /* ... */
  plugins: [
    /* ... */
    replace({
      /* ... */

>       'process.env.SAPPER_TIMESTAMP':

Reviewer 2: I think the docs mostly use tabs for code indentation, so should probably leave that for consistency. Also, I think this was easier to read before when this line and the next line were on one line

SAPPER_TIMESTAMP=$(date +%s%3N) npm run build
```

> When deploying to [Vercel], you can pass the environment variable into Now itself:

Reviewer 3: ```suggestion
When deploying to [Vercel], you can pass the environment variable into the Vercel cli tool itself:
```

SAPPER_TIMESTAMP=$(date +%s%3N) npm run build
```

When deploying to [Vercel], you can pass the environment variable into Now itself:

```bash
now -e SAPPER_TIMESTAMP=$(date +%s%3N)
```

> [vercel]: https://vercel.com/home

Reviewer 2: should this be `[Vercel]`? I don't know if it makes a difference. did you test rendering it?
